src/ivc/fold_relaxed_plonk_instance_chip.rs

```diff
@@ -242,15 +259,20 @@ where
         r: &[AssignedCell<C::Base, C::Base>],
     ) -> Result<Vec<AssignedPoint<C>>, Error> {
         let ecc = EccChip::<C, C::Base, T>::new(config.clone());
-        let results: Result<Vec<AssignedPoint<C>>, Error> = folded_W
+
+        folded_W
             .iter()
             .zip_eq(input_W_commitments)
-            .map(|(W1, W2)| -> Result<AssignedPoint<C>, Error> {
+            .enumerate()
+            .map(|(W_index, (W1, W2))| -> Result<AssignedPoint<C>, Error> {
                 let rW = ecc.scalar_mul(region, W2, r)?;
-                Ok(ecc.add(region, W1, &rW)?)
+                let res = ecc.add(region, W1, &rW)?;
+                debug!(
```

Comment:
NBD: I usually will remove debug code after testing.

Reply:
This is not `dbg!` this is `log::debug!` expression. The lowest level of logs, even lower `log::trace!`

It's off on a real running system because of verbosity, but for debugging some cases it can stay on. 

I left it here meaningfully

Reply:
This is just handy for projects under active development, to include the most detailed action log possible at any time, rather than adding debug messages every time. 

Such more generalised code for debugging, which at the TRACE level and above does not give any overhead